### ISSUE TITLE
Accessibility of adding songs to playlists is currently quite cumbersome with a screen reader

### ISSUE THREAD

**Issue by Vaalyn** [AzuraCast/AzuraCast#3488]
In order to add a song to a playlist a user of a screen reader has to first tab all the way through to a song, then select it and then tab all the way back to the top to the `Playlists` button.

We should improve this by adding a button to the actions column that allows one to add that song to a playlist.
In our Discord a user proposed that we could show a modal dialog like we do with the `Rename` button where a user can select the playlists that this song should be assigned to.

This might also be a good change for the `Playlists` button itself so that both use the same modal dialog.

**Comment by dingpengyu** [comment]
cc
@SlvrEagle23

**Comment by SlvrEagle23** [comment]
@dingpengyu No need to ping; I am aware of this issue.

**Comment by dingpengyu** [comment]
Thank you for your reply, I have deleted the comment posted above.

**Comment by SlvrEagle23** [comment]
@Vaalyn My proposed solution takes a bit of an "end-run" around the original problem by simply making it easier to set playlists for a single media item by foregoing the entire playlists dropdown and instead using the "Edit Media" modal dialog box to set the playlist while changing any other properties of the track.

This accomplishes the same purpose as having a per-track "set playlist" button but in a way that doesn't mess with the page's layout in general.

Another advantage is that the per-track playlist setting function has the existing playlists for the track already checked, which creates an experience that's probably closer to what users are expecting than the bulk "set playlists" function when editing a single track.

**Comment by Vaalyn** [comment]
I think it's okay and having a view were you can see all the playlists it is assigned to already checked is also a nice bonus.
When it comes to every day usability for visually impaired users I can't really judge it though. Would be awesome if someone who is affected by this could chime in here and share their thoughts on it.

**Comment by stickbear2015** [comment]
I'll need to mess with this some more, I'm just so used to checking what I want, hitting the playlist button, checking the playlist or playlists I want things in and smacking save, or is that submit? anyhow, I've never played in the dialog @SlvrEagle23 references. @Vaalyn once I get a test env up I'll have a look at this in development so I'm not breaking production boxes. It'll also give me a chance to digest what your proposing.
